### ISSUE TITLE
YouTube update breaking ff2mpv?

### ISSUE THREAD

**Issue by JoseHidalgo** [woodruffw/ff2mpv#104]
Hi all. It seems like the latest YouTube update is breaking ff2mpv. I'm now unable to stream YT videos from the browser to mpv.

Issue description:
- ff2mpv used to work perfectly until now, and nothing seems to have changed on my side.
- The problem appears in both Firefox and in Chrome (well, Brave Browser).
- I have tried with different YT videos, and the issue remains the same.
- Restarting the browser doesn't change anything.

Steps to reproduce:
- When I click on the ff2mpv icon, I see a CPU surge proving that something is happening... but MPV never opens.
- I can click as many times as I want on the icon, and I'll get lots of CPU surges... but MPV won't open anymore.

Additional info:
- To confirm the issue, I tried to download a YT video with JDownloader2. JD2 has always been successful at that. But today it wasn't even able to find the video! Which suggested that YT has indeed changed something.
- Even better, JD2 informed me that an update was available. After applying the update, I could immediately download the same YT video. So it looks like an easy fix. Hopefully a similar fix can be found for ff2mpv.

I'll be happy to provide more info if needed. Thanks!

**Comment by woodruffw** [comment]
Thank you for the report.

> Hi all. It seems like the latest YouTube update is breaking ff2mpv. I'm now unable to stream YT videos from the browser to mpv.

This is going to sound pedantic, so sorry in advance: YouTube *can't* break ff2mpv. All ff2mpv is is a small wrapper around `mpv` and `youtube-dl` (or `yt-dlp` or another fork); it isn't aware of either the site being streamed from (e.g. YouTube) or anything about how those tools attempt to stream it. 

If you're experiencing problems with YouTube, then it's possible that they've made a change that breaks the downloader backend you're using (`yt-dlp` or `youtube-dl`) or possibly something about MPV has changed.

You can test those theories with:

```
mpv your-url
# or youtube-dl
yt-dlp your-url
```

TL;DR: This isn't our bug. This should probably be reported upstream to either MPV or youtube-dl/yt-dlp (99% chance the latter, 1% chance the former).

For future reference: https://github.com/woodruffw/ff2mpv/wiki/Common-problems#urls-from-website-arent-working

I just realized that this information isn't included in the bug report template, so I'll fix that...

**Comment by woodruffw** [comment]
https://github.com/woodruffw/ff2mpv/commit/16f1f3e299791c238049a36f05d78277be8baace notes that this is the kind of thing we can't fix.

@JoseHidalgo can you confirm that it's just YouTube that's broken for you, and not all sites? If it's everything then it's our bug, but if it's just YouTube then it's something we need an upstream to fix.

**Comment by JoseHidalgo** [comment]
Thanks! It's OK, I understand, and you didn't sound pedantic at all :)

The problems seems YT-related indeed. I have tried playing a Vimeo video without issues.
Here's what Windows CMD is reporting with a random YT video:

![2023 08 15 - 18 12 38](https://github.com/woodruffw/ff2mpv/assets/10851603/cad0ab7a-054a-48dd-bd03-8d1741e210b8)

I guess I'll have to report it to MPV then, and hope for an update...

**Comment by JoseHidalgo** [comment]
I have just updated MPV (which in turn offers you to update YT-DLP), and it seems that the issue has been solved!

I should have thought of that first. Lesson learned for the future.

Problem solved, and sorry for the bother. :)

**Comment by woodruffw** [comment]
No problem at all -- thank you for resolving and confirming that it was an upstream break!
